packages/timeline-state-resolver/src/conductor.ts

```diff
@@ -657,19 +693,24 @@ export class Conductor extends EventEmitter<ConductorEvents> {
 		if (newDevice.initialized === true) {
 			throw new Error('Device ' + deviceId + ' is already initialized!')
 		}
-
 		this.emit(
 			'info',
 			`Initializing device ${newDevice.deviceId} (${newDevice.instanceId}) of type ${DeviceType[deviceOptions.type]}...`
 		)
-
-		await newDevice.init(deviceOptions.options, activeRundownPlaylistId)
-
-		await newDevice.reloadProps() // because the device name might have changed after init
-
-		this.emit('info', `Device ${newDevice.deviceId} (${newDevice.instanceId}) initialized!`)
-
-		return newDevice
+		return makeAbortable(async () => {
+			const throwIfAborted = () => {
+				if (options?.signal?.aborted) {
+					throw new AbortError(`Device "${deviceId}" initialisation aborted`)
+				}
+			}
+			throwIfAborted()
+			await newDevice.init(deviceOptions.options, activeRundownPlaylistId)
+			throwIfAborted()
+			await newDevice.reloadProps()
+			throwIfAborted()
```

Comment:
Is it really necesary to check if aborted this many times so close to each other?

Reply:
Yes because there's async code between those checks. Why would we wait for something we no longer need?

Reply:
It is not an elegant solution, but it is necessary.
A suggestion could be to create a reducer, that calls throwIfAborted before the first and after each statement/ action:

```typescript
async function runAbortableActions(actions: (() => Promise<void>)[], signal?: AbortSignal): Promise<void> {
  const throwIfAborted = async () => {
    if (!signal?.aborted) {
      return
    }
    throw new AbortError(`Device "${deviceId}" initialisation aborted.`)
  }
  await actions.reduce(
    (p, action) => p.then(action).then(throwIfAborted),
    throwIfAborted()
  )
}
```

Then this could be written as
```typescript
   ...
   runAbortableActions([
     () => newDevice.init(deviceOptions.options, activeRundownPlaylistId),
     () => newDevice.reloadProps(),
   ], options?.signal)
   ...
```
It creates some (unnecessary?) overhead, so @ianshade's solution might be the better choice.